prime-router/src/main/kotlin/transport/SoapTransport.kt

```diff
@@ -129,8 +129,10 @@ class SoapTransport(private val httpClient: HttpClient? = null) : ITransport {
         val receiver = header.receiver ?: error("No receiver defined for report ${header.reportFile.reportId}")
         // get the external file name to send to the client, if we need it
         val fileName = header.reportFile.externalName
+        // get the report id from the header
+        val headerReportId = header.reportFile.reportId
```

Comment:
Looks like the reportID is passed to the send function as property: **sentReportId**. Could you use that instead of creating a new variable?

Reply:
I could, and in theory it _should_ be the same, but there's no way for me to know if someone is passing in a different report id there for whatever reason. Part of the reason I went with creating the new variable is because on (formerly line 133) the existing code was pulling from the header.

Reply:
Oh yeah, it looks like there's a mixture of uses. I see the property used in actionHistory.trackSentReport() and actionHistory.trackItemLineages(). But then the header is used again on line 186. Seems like this is an opportunity to align on usage throughout the function. Then if you decide to only use the header version then property should just be removed.

Reply:
Can't remove the property because it's an overridden function.